### ISSUE TITLE
Uses legacy componentWillMount

### ISSUE THREAD

**Issue by jfrolich** [gaearon/react-side-effect#54]
This component does not work in concurrentMode

**Comment by lourd** [comment]
Hey Jaap, thanks for noticing and filing an issue!

I went ahead and started work at resolving this on the [`use-did-mount` branch](https://github.com/gaearon/react-side-effect/commit/879793a05c471f144b5e9d49423325a28fc08403). Can you please review it and also add a test case that makes sure the library is now concurrent mode compliant? Just fork from my branch.

Let me know if you have any questions. Thanks!

**Comment by jfrolich** [comment]
Not sure I have time to write a test, I don't use this library directly, just wanted to report it :)

**Comment by tyscorp** [comment]
Changing from `componentWillMount` to `componentDidMount` isn't enough as described here: #40

**Comment by seansfkelley** [comment]
This also triggers warnings when using `StrictMode`.

**Comment by mikecousins** [comment]
This now barfs warnings in React v16.9.

**Comment by lourd** [comment]
I'm gonna ping Dan on Twitter and ask if he can chime in here.

**Comment by gaearon** [comment]
I’ll take a look on Monday if I don’t forget. However this whole library doesn’t really make sense to me anymore so I’d encourage migrating away from these patterns. For example it wouldn’t work with a streaming renderer. Something using context would probably make more sense.

**Comment by mikecousins** [comment]
Yeah, it's used in react-document-title. So I guess we'll need to get away from using that or rewrite react-document-title away from react-side-effect. react-helmet also uses it.

**Comment by danielweck** [comment]
Looks like `react-helmet-async` is a useful alternative:
https://github.com/staylor/react-helmet-async

**Comment by danielweck** [comment]
Sorry, I posted my previous comment in the wrong repo! :)

**Comment by kud** [comment]
I'm here only because I also use `react-document-title`.

@mikecousins ' solution sounds the idea!

**Comment by verheyenkoen** [comment]
I see commit f75bc966aedcd136246836afd7289c4bc4496c25 already added the `UNSAFE_` prefix to the `componentWillMount` method to suppress the warning for this React major version. 

Not sure if this makes any sense but I just locally replaced that method name with `componentDidMount` (as suggested by the warning message) and all tests still pass. Could that be enough to fix this issue as a whole?

**Comment by lourd** [comment]
Unfortunately not, I believe there's just a coverage gap in the tests
regarding mounting order. There's discussion in another issue or PR where
we figured that out.

**Comment by codeaid** [comment]
So does this issue that was opened back in 2018 have any resolution? React helmet is throwing `Using UNSAFE_componentWillMount in strict mode is not recommended` and it seems it's because of this library.

**Comment by kud** [comment]
-> https://github.com/staylor/react-helmet-async
